lib/backend/k8s/syncer.go

```diff
@@ -379,7 +408,20 @@ func (syn *kubeSyncer) performSnapshot() ([]model.KVPair, map[model.Key]bool, re
 
 		for _, c := range confList {
 			snap = append(snap, *c)
-			keys[c.Key] = true
+			keys[fmt.Sprintf("%s", c.Key)] = true
+		}
+
+		// Sync IP Pools.
+		poolList, err := syn.kc.List(model.IPPoolListOptions{})
+		if err != nil {
```

Comment:
Ah I see the comment at the top of the loop.  Should we be checking for a specific error type here?  Or perhaps the message "Error syncing" is misleading.  It's more "Error querying pools - waiting for k8s datastore to initialize" or something like that.